youngeek-0410/sekigae
# feat: update User API #5

# Overview 
UserAPIの完成 (firebase auth loginの機能追加)
- create (修正)
- put (修正)
- read (既存)
- delete (既存)

firebase authを用いたGoogleログインの機能がなかったため追加。(開発時にswaggerでemail & password認証が必要なのでその機能は残してある。)

# Issue number 
Close #5 


# How to check the revision
1. setup environment variables files
2. add fastapi/firebase_credentials.json
3. up containers
4. access to swagger
5. check UserAPI (group User)

pythonのライブラリを追加したため`poetry install `が必要かもしれない。

# Points for Review 
- コード中のコメントにあるセキュリティ的問題

# Remarks 
- [ ] deleteが論理消去に依るUnique制約の問題は改めて別issueで再考して修正する
